### PR TITLE
Add Tag Length Constraint

### LINKED ISSUE #249: Tags can be too long
Add tag length limit

### PR DESCRIPTION
**Pull request by AerollingOverDeepBlueSeas**
Fixes #249

### CODE REVIEW

File: src/test/java/seedu/address/model/tag/TagTest.java [hunk 1]
```diff
@@ -15,6 +16,19 @@ public void constructor_null_throwsNullPointerException() {
     public void constructor_invalidTagName_throwsIllegalArgumentException() {
         String invalidTagName = "";
         assertThrows(IllegalArgumentException.class, () -> new Tag(invalidTagName));
+        // starts with -, illegal
+        assertThrows(IllegalArgumentException.class, () -> new Tag("-"));
+        // 31 characters - too long
+        assertThrows(IllegalArgumentException.class, () -> new Tag("abcdefghijklmnopqrstuvwxyzabcde"));
```
Comment (Hugo-Chia) [inline src/test/java/seedu/address/model/tag/TagTest.java:22]: I like that the test are written at the boundary cases
